### PR TITLE
Abstracts metamask functionality. Introduces wallet interface.

### LINKED ISSUE #8: Support for Multiple Wallets
**Is your feature request related to a problem?**

As a dapp developer that has to integrate several different wallets and advise users how to interact with them, I would like to have one testing solution that let's me test my integration with different wallets to ensure wallet functionality and UX (User Experience) is aligned with what I deliver to users.

**Describe the solution you'd like**

To introduce some kind of adapter pattern to dAppwright. This would specify an interface of common interactions (ie. switch networks, add account, approve transaction, etc) that a wallet adapter would use to define functionality. 

As a developer, I would be able to keep the testing code in my test suite but use a `before` hook to set the desired wallet to verify a flow.

```mermaid
flowchart TD
    subgraph Playwright Tests
    suite[Test Suite] --> before["before#40;#41; - dAppwright.Launch#40;wallet#41;"] --> test["test() - dAppwright.SignMessage#40;#41;"]
    end
    subgraph dAppwright
    test --> router[router]
    router -- or --> coinbase["Coinbase Wallet Adapter"]
    router -- or --> metamask[Metamask Adapter]
    coinbase --> coinbaseAction["perform SignMessage"]
    metamask --> metamaskAaction["perform SignMessage"]
    end

```

**Describe alternatives you've considered**

There are many libraries attempting to bridge the gap of integrating with wallets but not from a testing perspective. I personally haven't been able to find an alternative in the market beyond waiting for another library like [Synpress](https://github.com/Synthetixio/synpress) to adopt a pattern like this as well, but that solution is on a different platform.

### PR DESCRIPTION
**Pull request by osis**
**Short description of work done**
Info is in https://github.com/TenKeyLabs/dappwright/issues/8

### PR Checklist

- [x] I have run linter locally
- [x] I have run unit and integration tests locally
- [x] Update configuration the newest version (readme and const)
- [x] Rebased to master branch / merged master

### Changes
This abstracts the Metamask functionality in the project and introduces an interface so that other wallet tech can implements the common dAppwright actions.

Closes #8

### CODE REVIEW

File: src/wallets/metamask/actions/helpers/selectors.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  );
+
+export const getErrorMessage = async (page: Page): Promise<string | false> => {
+  const options: Parameters<Page['waitForSelector']>[1] = { timeout: 1000 };
+
+  const errorElement = await Promise.race([
+    page.waitForSelector(`span.error`, options),
+    page.waitForSelector(`.typography--color-error-1`, options),
+    page.waitForSelector(`.typography--color-error-default`, options),
+    // page.waitForSelector(`.form-field__input--warning`, options),
```
Comment (RayzorFlash) [inline src/wallets/metamask/actions/helpers/selectors.ts:18]: Can be removed?

File: src/wallets/metamask/setup/downloader.ts [hunk 2]
```diff
@@ -14,18 +17,73 @@ export type Path =
...
+      console.log(); // new line
+      if (version === 'latest')
+        console.warn(
+          '\x1b[33m%s\x1b[0m',
+          `It is not recommended to run metamask with "latest" version. Use it at your own risk or set to the recommended version "${recommendedVersion}".`,
+        );
+      else if (isNewerVersion(recommendedVersion, version))
+        console.warn(
+          '\x1b[33m%s\x1b[0m',
+          `Seems you are running newer version (${version}) of MetaMask that recommended by dappwright team.
```
Comment (RayzorFlash) [inline src/wallets/metamask/setup/downloader.ts:44]:
```suggestion
          `Seems you are running a newer version (${version}) of MetaMask than recommended by the Dappwright team.
```

File: src/wallets/metamask/setup/downloader.ts [hunk 3]
```diff
@@ -14,18 +17,73 @@ export type Path =
...
+      Use it at your own risk or set to the recommended version "${recommendedVersion}".`,
+        );
+      else if (isNewerVersion(version, recommendedVersion))
+        console.warn(
+          '\x1b[33m%s\x1b[0m',
+          `Seems you are running older version (${version}) of MetaMask that recommended by dappwright team.
```
Comment (RayzorFlash) [inline src/wallets/metamask/setup/downloader.ts:50]:
```suggestion
          `Seems you are running an older version (${version}) of MetaMask than recommended by the Dappwright team.
```

File: test/test.spec.ts [hunk 4]
```diff
@@ -131,12 +123,12 @@ describe('dappwright', () => {
...
 
   it('should return token balance', async () => {
-    const tokenBalance: number = await metamask.helpers.getTokenBalance('ETH');
+    const tokenBalance: number = await metamask.getTokenBalance('ETH');
     expect(tokenBalance).to.be.greaterThan(0);
   });
 
   it('should return 0 token balance when token not found', async () => {
-    const tokenBalance: number = await metamask.helpers.getTokenBalance('FARTBUCKS');
+    const tokenBalance: number = await metamask.getTokenBalance('FARTBUCKS');
```
Comment (RayzorFlash) [inline test/test.spec.ts:131]: Interesting test token name 😂

**Review by RayzorFlash — APPROVED** [review]
Just minor cleanup to make some verbage more professional. Otherwise LGTM

File: test/test.spec.ts (hunk 4 above)
Comment (osis) [reply test/test.spec.ts:131]: LOL that wasn't me...

**Pull request merged**